File: universal7885-common/apps/aidl-support/libaidlvintf/AIDLInterface.kt
package com.royna.aidlvintf

object AIDLInterface {
     fun makeAIDLStr(name: String, kInterface: String, instance: String = "default") : String {
          var local = kInterface
          if (!kInterface.startsWith("I")) local = "I" + kInterface
          return name + "." + local + "/" + instance
     }
}

File: universal7885-common/apps/aidl-support/libaidlvintf/AIDLInterface.cpp
#include <sstream>
#include "AIDLInterface.h"

constexpr const char *kInterfacePrefix = "I";

std::string makeAIDLInterfaceString(const struct aidl_vintf *info)
{
	std::stringstream kInterfaceStrBuilder;
	kInterfaceStrBuilder << info->name;
	kInterfaceStrBuilder << ".";
	if (info->interface.rfind(kInterfacePrefix, 0) != 0)
	{
		kInterfaceStrBuilder << kInterfacePrefix;
	}
	kInterfaceStrBuilder << info->interface;
	kInterfaceStrBuilder << "/" << info->instance;
	return kInterfaceStrBuilder.str();
}
